cli/tests/test_observability_v8_upgrade_migration.py

```
@@ -71,6 +72,25 @@ def _begin_verified_upgrade_receipt(self, target_version: str = "9.9.9") -> str:
             )
         )
 
+    def test_target_bundle_subprocess_uses_isolated_python(self) -> None:
+        observed: list[str] = []
+
+        def complete(command, **_kwargs):
```

Ruff (0.15.21)

[warning] Missing return type annotation for private function `complete`

(ANN202)